### PR TITLE
Packaging: Refactor java and C#

### PR DESCRIPTION
**Pull request by aeisenberg**
Refactors two languages at the same time, similar to how C++ was done.

### CODE REVIEW

File: csharp/ql/examples/qlpack.yml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 name: codeql-csharp-examples
 version: 0.0.0
-libraryPathDependencies: codeql-csharp
+dependencies:
+  codeql/csharp-all: 0.0.1
```
Comment (adityasharad) [inline csharp/ql/examples/qlpack.yml:4]: Should this be 0.0.2, or ^0.0.1?

Comment (aeisenberg) [reply csharp/ql/examples/qlpack.yml:4]: Good catch. That should be made more flexible.

**Comment by aeisenberg** [comment]
Could someone explain why the qldocs are failing? It looks like neither the baseline nor this PR has those items documented.

**Review by hvitved — COMMENTED** [review]
I have really no idea how to review this PR, but I do note that none of the contents in the C# `.qll` files change, which is good.

**Review by adityasharad — APPROVED** [review]

**Comment by adityasharad** [comment]
Planning to ignore the QLDoc Checks, since they are confused by the files being moved. Will merge once the Java and C# tests both pass.

**Pull request merged**